.evergreen/auth_oidc/azure/test.py

```diff
@@ -3,45 +3,48 @@
 import json
 from urllib.request import urlopen, Request
 from pymongo.auth import _AUTH_MAP, _authenticate_oidc
+from pymongo.auth_oidc import OIDCCallback, OIDCCallbackContext, OIDCCallbackResult
 
 # Force MONGODB-OIDC to be enabled.
 _AUTH_MAP["MONGODB-OIDC"] = _authenticate_oidc
 
 app_id = os.environ['AZUREOIDC_CLIENTID']
 object_id = os.environ['AZUREOIDC_USERNAME']
 
-def callback(client_info, server_info):
-    url = "http://169.254.169.254/metadata/identity/oauth2/token"
-    url += "?api-version=2018-02-01"
-    url += f"&resource=api://{app_id}"
-    url += f"&object_id={object_id}"
-    headers = { "Metadata": "true", "Accept": "application/json" }
-    request = Request(url, headers=headers)
-    try:
-        with urlopen(request, timeout=server_info['timeout_seconds']) as response:
-            status = response.status
-            body = response.read().decode('utf8')
-    except Exception as e:
-        msg = "Failed to acquire IMDS access token: %s" % e
-        raise ValueError(msg)
-
-    if status != 200:
-        print(body)
-        msg = "Failed to acquire IMDS access token."
-        raise ValueError(msg)
-    try:
-        data = json.loads(body)
-    except Exception:
-        raise ValueError("Azure IMDS response must be in JSON format.")
+class MyCallback(OIDCCallback):
+    def fetch(self, context: OIDCCallbackContext) -> OIDCCallbackResult:
+        url = "http://169.254.169.254/metadata/identity/oauth2/token"
+        url += "?api-version=2018-02-01"
+        url += f"&resource=api://{app_id}"
+        url += f"&object_id={object_id}"
+        headers = { "Metadata": "true", "Accept": "application/json" }
+        print('Fetching url', url)
```

Comment:
```suggestion
        print('Fetching url', url)
```

Do we still want this print statement?

Reply:
Yes, for debugging, it isn't printing anything sensitive.